Objective-C/CBLLog.mm

```diff
@@ -149,69 +50,40 @@ static void sendToCallbackLogger(C4LogDomain d, C4LogLevel l, NSString* message)
 - (instancetype) initWithDefault {
     self = [super init];
     if (self) {
-        // The most default callback log level:
-        C4LogLevel callbackLogLevel = kC4LogWarning;
+        // Initialize new logging system
+        [CBLLogSinks init];
         
-#ifdef DEBUG
-        // Check if user overrides the default callback log level:
-        NSString* userLogLevel = [NSUserDefaults.standardUserDefaults objectForKey: @"CBLLogLevel"];
-        if (userLogLevel) {
-            callbackLogLevel = string2level(userLogLevel);
-        }
-        if (callbackLogLevel != kC4LogWarning) {
-            NSLog(@"CouchbaseLite minimum log level is %s", kLevelNames[callbackLogLevel]);
-        }
-#endif
-        
-        // Enable callback logging:
-        c4log_writeToCallback(callbackLogLevel, &logCallback, true);
-        
-        // Set log level for each domains to the lowest:
-        kCBL_LogDomainDatabase  = setNamedLogDomainLevel("DB", kC4LogDebug);
-        kCBL_LogDomainQuery     = setNamedLogDomainLevel("Query", kC4LogDebug);
-        kCBL_LogDomainSync      = setNamedLogDomainLevel("Sync", kC4LogDebug);
-        kCBL_LogDomainWebSocket = setNamedLogDomainLevel("WS", kC4LogDebug);
-        kCBL_LogDomainListener  = setNamedLogDomainLevel("Listener", kC4LogDebug);
-        setNamedLogDomainLevel("BLIP", kC4LogDebug);
-        setNamedLogDomainLevel("SyncBusy", kC4LogDebug);
-        setNamedLogDomainLevel("TLS", kC4LogDebug);
-        setNamedLogDomainLevel("Changes", kC4LogDebug);
-        setNamedLogDomainLevel("Zip", kC4LogDebug);
-        setNamedLogDomainLevel("BLIPMessages", kC4LogDebug);
-        
-#ifdef DEBUG
-        // Now map user defaults starting with CBLLog... to log levels:
-        NSDictionary* defaults = [NSUserDefaults.standardUserDefaults dictionaryRepresentation];
-        for (NSString* key in defaults) {
-            if ([key hasPrefix: @"CBLLog"] && ![key isEqualToString: @"CBLLogLevel"]) {
-                const char *domainName = key.UTF8String + 6;
-                if (*domainName == 0)
-                    domainName = "Default";
-                C4LogDomain domain = c4log_getDomain(domainName, true);
-                C4LogLevel level = string2level(defaults[key]);
-                c4log_setLevel(domain, level);
-                NSLog(@"CouchbaseLite logging to %s domain at level %s", domainName, kLevelNames[level]);
-            }
-        }
-#endif
-        
-        // Keep the current callback log level:
-        _callbackLogLevel = (CBLLogLevel)callbackLogLevel;
+        [CBLLogSinks setVAPI: LogAPINone];
```

Comment:
here

Reply:
It is better do be done in `[CBLLogSinks init]` so the logic is controlled in a single place and done only once.